api/v1alpha1/api.go

```diff
@@ -124,42 +124,42 @@ type LLMRouteRuleMatch struct {
 
 // +kubebuilder:object:root=true
 
-// LLMBackend is a resource that represents a single backend for LLMRoute.
+// AIBackend is a resource that represents a single backend for AIRoute.
 // A backend is a service that handles traffic with a concrete API specification.
 //
-// A LLMBackend is "attached" to a Backend which is either a k8s Service or a Backend resource of the Envoy Gateway.
+// A AIBackend is "attached" to a Backend which is either a k8s Service or a Backend resource of the Envoy Gateway.
 //
-// When a backend with an attached LLMBackend is used as a routing target in the LLMRoute (more precisely, the
-// HTTPRouteSpec defined in the LLMRoute), the ai-gateway will generate the necessary configuration to do
+// When a backend with an attached AIBackend is used as a routing target in the AIRoute (more precisely, the
+// HTTPRouteSpec defined in the AIRoute), the ai-gateway will generate the necessary configuration to do
 // the backend specific logic in the final HTTPRoute.
-type LLMBackend struct {
+type AIBackend struct {
```

Comment:
AIBackend is too broad, suggests AIServiceBackend.

Reply:
+1 for AIServiceBackend

Reply:
how about AIProvider (we tend to use in discussion)?

Reply:
AIServiceBackend wfm as well

Reply:
> how about AIProvider (we tend to use in discussion)?

we are still not supporting provider specific configurations here such as supported models, supported authenticated types, transformation etc. for now i feel like this more towards to managing backend configuration